apps/bestofjs-nextjs/src/components/tags/project-tag-hover-card.tsx

```diff
@@ -0,0 +1,147 @@
+"use client";
+
+import { ReactNode, useState } from "react";
+import useSWR, { SWRConfiguration } from "swr";
+
+import { ProjectAvatar } from "@/components/core";
+
+import { TagIcon } from "../core";
+import {
+  HoverCard,
+  HoverCardContent,
+  HoverCardTrigger,
+} from "../ui/hover-card";
+import { Skeleton } from "../ui/skeleton";
+
+export const ProjectTagHoverCard = ({
+  children,
+  tag,
+}: {
+  children: ReactNode;
+  tag: BestOfJS.Tag;
+}) => {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+
+  const onOpenChange = (isOpen: boolean) => {
+    setIsOpen(isOpen);
+  };
+
+  return (
+    <HoverCard onOpenChange={onOpenChange}>
+      <HoverCardTrigger asChild>{children}</HoverCardTrigger>
+      <HoverCardContent className="w-80">
+        <TagCardContent description={tag.description} name={tag.name}>
+          <CardTagProjects isOpen={isOpen} tag={tag} />
+        </TagCardContent>
+      </HoverCardContent>
+    </HoverCard>
+  );
+};
+
+const CardTagProjects = ({
+  isOpen,
+  tag,
+}: {
+  isOpen: boolean;
+  tag: BestOfJS.Tag;
+}) => {
+  if (!isOpen) return;
+
+  return (
+    <div className="flex-col">
+      <div className="py-2 text-left" style={{ color: "white" }}>
+        <h1>Top 5 Projects</h1>
+      </div>
+      <div>
+        <FetchTagProjects tag={tag} />
+      </div>
+    </div>
+  );
+};
+
+const TagCardContent = ({
+  children,
+  description,
+  name,
+}: {
+  children: ReactNode;
+  description: string | undefined;
+  name: string;
+}) => (
+  <div className="flex justify-between space-x-4">
+    <div className="w-[100%] space-y-1">
+      <div className="flex items-center space-x-1">
+        <div style={{ color: "#F59E0B" }}>
+          <TagIcon />
+        </div>
+        <h4 className="text-sm font-semibold">{name}</h4>
+      </div>
+      <p className="text-left text-sm">{description}</p>
+      <div>
+        <div className="grow text-sm text-muted-foreground">{children}</div>
+      </div>
+    </div>
+  </div>
+);
+
+function getTagProjecListStyles({ isFirst }: { isFirst: boolean }) {
+  return {
+    color: "white",
+    gap: 10,
+    alignItems: "center",
+    borderStyle: "solid",
+    borderColor: `rgba(255, 255, 255, 0.2)`,
+    borderBottomWidth: "1px",
+    borderTopWidth: isFirst ? 1 : 0,
+    padding: "8px 0",
+  };
+}
```

Comment:
I guess it's just for the prototype as I can see you re-used what we did for OG images generated on the server but we should not use inline styles in the web app, especially because we support light and dark modes.

Reply:
In this case is it better to stick with tailwind utility class names then?

Reply:
Since you mentioned wanting to try and improve the look and feel on your end, I will leave this part up to you for now 😄